snapshots/overlay/overlay.go

```diff
@@ -518,3 +624,53 @@ func supportsIndex() bool {
 	}
 	return false
 }
+
+// supportsMetacopy checks whether the "metacopy" option is supported by the kernel.
+func supportsMetacopy() bool {
+	if _, err := os.Stat("/sys/module/overlay/parameters/metacopy"); err != nil {
+		logrus.Infof("Metacopy unsupported.")
+		return false
+	}
+	return true
+}
+
+// isMetacopyOn checks whether the "metacopy" option is on/off by kernel.
+// the caller should call supportsMetacopy first. isMetacopyOn returns true
+// if "metacopy=on", if "metacopy=off" or metacopy not supported it returns false.
+func isMetacopyOn() bool {
+	content, err := os.ReadFile("/sys/module/overlay/parameters/metacopy")
```

Comment:
We can OOM here if the file content is changed to be very big.

Of course, if other parts of the code in containerd don't care about it, I think we shouldn't care here either. But I don't if we generally care about this in containerd or not :)

Reply:
I got it, but not sure this may be a real case. If this file will be too large it means the system is about to crush or we don't have memory to keep read file content we'll just return false, may be  it is needed to return error too

Reply:
Ouch, I thought this was when reading the file we create. Sorry :D